### PR TITLE
Refactor and Removed some unnecessary code.

### PR DESCRIPTION
**Pull request by ishan16696**
**What this PR does / why we need it**:
1. This PR removes the `EtcdConnectionConfig struct` from `pkg/etcdutil` which I forgot to remove while rebasing [leader-election PR](https://github.com/gardener/etcd-backup-restore/pull/353) as [EtcdConnectionConfig struct](https://github.com/gardener/etcd-backup-restore/blob/322d3c74fcd5ea7be9961ce5e9d7ffed48ecc34e/pkg/types/etcdconnection.go#L37) already present in `pkg/types` .
2. This PR also use the [--etcd-defrag-timeout](https://github.com/gardener/etcd-backup-restore/blob/322d3c74fcd5ea7be9961ce5e9d7ffed48ecc34e/pkg/types/etcdconnection.go#L73) flag in Defragmentation which I missed to use in [leaderElection PR](https://github.com/gardener/etcd-backup-restore/pull/353/files#diff-335b2df327a203ec96e960dd7fe076e4f87631ecc69a366cb595d330db2dc0fcR175) 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by aaronfern — CHANGES_REQUESTED** [review]
Thanks for the PR @ishan16696!

Just one small concern from me. Otherwise it looks really good

File: pkg/compactor/compactor.go [hunk 1]
```diff
@@ -144,9 +144,7 @@ func (cp *Compactor) Compact(opts *brtypes.CompactOptions) (*brtypes.Snapshot, e
 		}
 		defer client.Close()
 
-		defragCtx, defragCancel := context.WithTimeout(ctx, opts.DefragTimeout.Duration)
-		err = etcdutil.DefragmentData(defragCtx, clientMaintenance, client, ep, cp.logger)
-		defragCancel()
+		err = etcdutil.DefragmentData(ctx, clientMaintenance, client, ep, opts.DefragTimeout.Duration, cp.logger)
```
[the remark this answers is not in the document]
Comment (ishan16696) [reply pkg/compactor/compactor.go]: Addressed. PTAL

File: pkg/types/etcdconnection.go [hunk 2]
```diff
@@ -31,6 +31,9 @@ const (
 	DefaultDefragConnectionTimeout time.Duration = 8 * time.Minute
 	// DefaultSnapshotTimeout defines default timeout duration for taking FullSnapshot.
 	DefaultSnapshotTimeout time.Duration = 8 * time.Minute
+
+	// DefragRetryPeriod used for ETCD defrag retry.
```
Comment (timuthy) [inline pkg/types/etcdconnection.go:35]:
```suggestion
	// DefragRetryPeriod is used as the duration after which a defragmentation is retried. 
```

File: pkg/compactor/compactor.go [hunk 3]
```diff
@@ -144,9 +142,9 @@ func (cp *Compactor) Compact(opts *brtypes.CompactOptions) (*brtypes.Snapshot, e
 		}
 		defer client.Close()
 
-		defragCtx, defragCancel := context.WithTimeout(ctx, opts.DefragTimeout.Duration)
-		err = etcdutil.DefragmentData(defragCtx, clientMaintenance, client, ep, cp.logger)
-		defragCancel()
+		defragCtx, defragCtxCancel := context.WithCancel(ctx)
+		defer defragCtxCancel()
```
Comment (timuthy) [inline pkg/compactor/compactor.go:146]: What's the benefit/need to create a separate context here?

Comment (ishan16696) [reply pkg/compactor/compactor.go]: removed

**Review by timuthy — APPROVED** [review]
/lgtm

**Review by aaronfern — APPROVED** [review]
/lgtm

**Comment by abdasgupta** [comment]
Please merge it @ishan16696

**Pull request merged**